Bug: findIds() crashes with StringIndexOutOfBoundsException in RedissonLiveObject

### Redis version

6.2.7

### Redisson version

4.3.1

### Redisson configuration

Default config

### What is the Expected behavior?

`findIds(TestIndexed1.class)` returns all persisted object IDs regardless of how many objects exist or how many cluster nodes are in use:

```java
RLiveObjectService s = redisson.getLiveObjectService();
TestIndexed1 t1 = new TestIndexed1(); t1.setId("1");
TestIndexed1 t2 = new TestIndexed1(); t2.setId("2");
s.persist(t1, t2);
Iterable<String> ids = s.findIds(TestIndexed1.class);
assertThat(ids).containsExactlyInAnyOrder("1", "2");
```

Works equally with 2 objects, 25 objects, single-node, or 3-node cluster.

### What is the Actual behavior?

`findIds()` throws `RedisException` wrapping `StringIndexOutOfBoundsException` when more than ~10 LiveObjects of the same class exist (single-node pagination) **or** when running against a cluster with more than 1 master node:

```
org.redisson.client.RedisException: Unexpected exception while processing command
Caused by: java.lang.StringIndexOutOfBoundsException: Range [0, -1) out of bounds for length 1
```

### Root Cause

```java
// Broken: index shared across ALL SCAN pages
new Convertor<Object>() {
    int index;
    public Object convert(Object obj) {
        index++;
        if (index == 1) { return obj; }  // Only skips cursor on page 1
        return namingScheme.resolveId(obj.toString()); // Cursor "0" on page 2 → crash
    }
}
```

The Convertor is called per-element (cursor, then each key). On page 2+, `index > 1`, so `resolveId("0")` runs on the cursor string `"0"`, which has no `{` braces → `substring(-1+1, -1)` → `StringIndexOutOfBoundsException`.

### Stack Trace
```
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 2.930 s <<< FAILURE! -- in org.redisson.RedissonLiveObjectServiceTest
[ERROR] org.redisson.RedissonLiveObjectServiceTest.testFindIdsPagination -- Time elapsed: 0.809 s <<< ERROR!
org.redisson.client.RedisException: Unexpected exception while processing command
        at org.redisson.RedissonLiveObjectServiceTest.testFindIdsPagination(RedissonLiveObjectServiceTest.java:1557)
Caused by: java.lang.StringIndexOutOfBoundsException: Range [0, -1) out of bounds for length 1

[ERROR] org.redisson.RedissonLiveObjectServiceTest.testFindIds -- Time elapsed: 0.134 s <<< ERROR!
org.redisson.client.RedisException: Unexpected exception while processing command
        at org.redisson.RedissonLiveObjectServiceTest.testFindIds(RedissonLiveObjectServiceTest.java:1539)
Caused by: java.lang.StringIndexOutOfBoundsException: Range [0, -1) out of bounds for length 1
```

### Additional information

_No response_

--- a/redisson/src/main/java/org/redisson/RedissonLiveObjectService.java
+++ b/redisson/src/main/java/org/redisson/RedissonLiveObjectService.java
@@ -32,13 +32,7 @@ import org.redisson.api.*;
 import org.redisson.api.annotation.*;
 import org.redisson.api.condition.Condition;
 import org.redisson.client.codec.StringCodec;
-import org.redisson.client.protocol.RedisCommand;
 import org.redisson.client.protocol.RedisCommands;
-import org.redisson.client.protocol.convertor.Convertor;
-import org.redisson.client.protocol.decoder.ListMultiDecoder2;
-import org.redisson.client.protocol.decoder.ListScanResult;
-import org.redisson.client.protocol.decoder.ListScanResultReplayDecoder;
-import org.redisson.client.protocol.decoder.ObjectListReplayDecoder;
 import org.redisson.command.CommandAsyncExecutor;
 import org.redisson.command.CommandBatchService;
 import org.redisson.liveobject.LiveObjectSearch;
@@ -759,21 +753,11 @@ public class RedissonLiveObjectService implements RLiveObjectService {
         NamingScheme namingScheme = commandExecutor.getObjectBuilder().getNamingScheme(entityClass);
         String pattern = namingScheme.getNamePattern(entityClass);
         RedissonKeys keys = new RedissonKeys(commandExecutor);
-
-        RedisCommand<ListScanResult<String>> command = new RedisCommand<>("SCAN",
-                new ListMultiDecoder2(new ListScanResultReplayDecoder(), new ObjectListReplayDecoder<Object>()), new Convertor<Object>() {
-            int index;
-            @Override
-            public Object convert(Object obj) {
-                index++;
-                if (index == 1) {
-                    return obj;
-                }
-                return namingScheme.resolveId(obj.toString());
-            }
-        });
-
-        return keys.getKeysByPattern(command, pattern, 0, count, null);
+        List<K> result = new ArrayList<>();
+        for (String name : keys.getKeysByPattern(pattern, count)) {
+            result.add((K) namingScheme.resolveId(name));
+        }
+        return result;
     }
 
     @Override

--- a/redisson/src/test/java/org/redisson/RedissonLiveObjectServiceTest.java
+++ b/redisson/src/test/java/org/redisson/RedissonLiveObjectServiceTest.java
@@ -1540,6 +1540,24 @@ public class RedissonLiveObjectServiceTest extends RedisDockerTest {
     }
 
     @Test
+    public void testFindIdsPagination() {
+        RLiveObjectService s = redisson.getLiveObjectService();
+        int count = 25;
+        List<TestIndexed1> entities = new ArrayList<>();
+        List<String> expectedIds = new ArrayList<>();
+        for (int i = 0; i < count; i++) {
+            TestIndexed1 t = new TestIndexed1();
+            t.setId("pg-" + i);
+            entities.add(t);
+            expectedIds.add("pg-" + i);
+        }
+        s.persist(entities.toArray(new TestIndexed1[0]));
+
+        Iterable<String> ids = s.findIds(TestIndexed1.class);
+        assertThat(ids).containsExactlyInAnyOrderElementsOf(expectedIds);
+    }
+
+    @Test
     public void testMergeList2() {
         RLiveObjectService s = redisson.getLiveObjectService();
         TestIndexed1 t1 = new TestIndexed1();

